----

# 회귀 모델 적합성 검증

참고 : 데이터 분석 준전문가 한 권으로 끝내기

Summary 함수를 통해 결정계수, F통계량, 잔차의 표준오차 등 주요 통계량의 정보를 출력할 수 있다.

## 1. Regression Analysis

In [2]:
# 난수 생성기의 시드 설정. 
set.seed(2) 

In [2]:
u <- runif(10, 0, 11)
v <- runif(10, 11, 20)
w <- runif(10, 1, 30)

u;v;w

[1] 7.87428298 0.24529171 0.49276717 8.01608060 1.21479267 9.13439729
 [7] 0.65194100 0.09117142 9.87657584 9.86208962

[1] 16.15200 19.12523 16.08198 18.76564 13.35264 15.73600 18.40019 13.37232
 [9] 12.03049 12.55313

[1]  3.451476 20.206888 29.443298 26.471105  4.929695 18.572989  2.341201
 [8]  4.721583 10.515000 15.108903

In [4]:
y = 3 + 0.1*u + 2*v -3*rnorm(10, 0, 0.1)
y

[1] 36.35949 41.24569 35.35031 41.57348 29.66499 35.66222 39.72315 30.09292
 [9] 28.11384 28.48200

- **runif(10, 0, 11)** : 평균이 0, 표준편차가 11인 연속균등분포로 부터 난수 10개를 추출하는 기능
- **rnorm(10,0,0.2)** : 평균이 0, 표준편차가 2인 정규분포로부터 난수 10개를 추출

## 2. Covert to DataFrame

In [7]:
dfrm <- data.frame(u,v,w,y)
dfrm

u          v        w         y       
1  7.87428298 16.15200  3.451476 36.35949
2  0.24529171 19.12523 20.206888 41.24569
3  0.49276717 16.08198 29.443298 35.35031
4  8.01608060 18.76564 26.471105 41.57348
5  1.21479267 13.35264  4.929695 29.66499
6  9.13439729 15.73600 18.572989 35.66222
7  0.65194100 18.40019  2.341201 39.72315
8  0.09117142 13.37232  4.721583 30.09292
9  9.87657584 12.03049 10.515000 28.11384
10 9.86208962 12.55313 15.108903 28.48200

## 3. Regression Analysis

In [8]:
lm(y~u+v+w, data=dfrm)


Call:
lm(formula = y ~ u + v + w, data = dfrm)

Coefficients:
(Intercept)            u            v            w  
  2.6011057    0.1002781    2.0276576    0.0004123  


- **Intercept** : 상수


그러므로 회귀식은 아래와 같다. 
$$ y = 2.6011 + 0.1002\times{u} + 2.0276\times{v} + 0.0004\times{w} $$

----